### ISSUE TITLE
calories.active type in Android not works well

### ISSUE THREAD

**Issue by juanico18** [dariosalvi78/cordova-plugin-health#295]
The plugin a bug on health.js on queryAggregated for type calories.active 
```javascript
Health.prototype.queryAggregated = function (opts, onSuccess, onError) {
  if (opts.dataType == 'calories.active') {
    //get basal average
    navigator.health.queryAggregated({
      dataType: 'calories.basal',
      endDate: opts.endDate,
      startDate: opts.startDate
    }, function (data) {
      var basal_ms = data.value / (opts.endDate - opts.startDate);
      //now get the total
      opts.dataType = 'calories';
      navigator.health.queryAggregated(opts, function (retval) {
       //and remove the basal
        retval.value -= basal_ms * (retval.endDate.getTime() - retval.startDate.getTime());
        //although it shouldn't happen....
        if (retval.value < 0) retval.value = 0;
        onSuccess(retval);
      }, onError);
    }, onError);
  }
``` 

  
 If opts comes with the bucket: 'day' param. retval is an array and not and object.
 
 I have fixed that by requesting from our app calories and calories.basal and adapting the code (But)

(But)->  the way active calories are obtained is not correct.
Gets the user's average basal calories.
It was consulted a year ago. GoogleFit returns 0 if the user set their weight and height more than 1 year ago. My case.
If it returns 0 the calculation is wrong.
Subtract the basal calories obtained from the total calories.

There would be another way to obtain active calories which is by obtaining activities
But to obtain the calories the plugin requests calorie and distance permissions. Making it necessary for the app to request location permissions.
It would be enough if the plugin got only the calories from the activities and not the distance.

So instead of return calories.active based on substract basal from total calories. We can get the calories from the activitys during that period.

Other thoughts:
GoogleFit should be replaced by HealthConnect.
Other option is to try the capacitor plugin https://github.com/perfood/capacitor-google-fit
But I think is not the calories from activities
Additionally, we want to migrate to capacitor 5 and it is not supported yet.

Final thoughts. I see two options to fix it.

1 Allow get calories from activities without request distance or
2 Change logic of obtain calories.active and instead of query basal. Query activities with calories.

I think this topic is related with these issues:
https://github.com/dariosalvi78/cordova-plugin-health/issues/98
https://github.com/dariosalvi78/cordova-plugin-health/issues/194
https://github.com/dariosalvi78/cordova-plugin-health/issues/216
https://github.com/dariosalvi78/cordova-plugin-health/issues/294

**Comment by dariosalvi78** [comment]
you are right, this doesn't work with buckets! Probably a use case that has never been tested. I will open a separate issue for this.

Unfortunately there is no "active calories" (calories added to the basal metabolic activity) in Google Fit, so this approximation is the best we can do. Basal calories are only estimated when setting weight, and we cannot foresee when this happens, neither is there a way (that I know of) to know when the last basal was computed. So we query 1 year and hope for the best.

Regarding activities, the calories associated to those are total, not active, so that wouldn't work either. If you look at the source code, it just queries for "calories" over the period of each activity.

Regarding Health connect, yes, it's on the TODO, there's another issue for this.

Capacitor: this plugin is for Cordova, I don't intend to support capacitor. If you want to do it, be my guest.


In summary: there is no way to get active calories in Google Fit as in Healthkit, what's done here is the best compromise I could think of. My suggestion is to avoid using active calories and go with the total ones.

Unless you have a better proposal, I will need to close the issue.

**Comment by dariosalvi78** [comment]
separate issue created for the bucketed active calories: https://github.com/dariosalvi78/cordova-plugin-health/issues/296

**Comment by juanico18** [comment]
@dariosalvi78 About health connect and capacitor. I just mentioned as other possibilities.
As we are already working with capacitor and Google Fit is going to be deprecated. I think that a capacitor plugin that connect with HealthConnect and with Apple Health covering same use cases than this plugin would be a good solution to maintain in the future
